crates/store/src/genesis/config/mod.rs

```diff
@@ -0,0 +1,291 @@
+            StorageMode::Private => AccountStorageMode::Private,
+            StorageMode::Public => AccountStorageMode::Public,
+        }
+    }
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+struct AssetEntry {
+    symbol: String, // TODO move to a wrapper around `TokenSymbol`
+    amount: u64,    // TODO we might want to provide `humantime`-like denominations
+}
+
+#[derive(Debug, Clone)]
+pub struct AccountFileWithName {
+    pub name: String,
+    pub account_file: AccountFile,
+}
+
+/// Secrets generated during the state generation
+#[derive(Debug, Clone)]
+pub struct AccountSecrets {
+    // name, account, private key, account seed
+    pub secrets: Vec<(Option<String>, Account, SecretKey, Word)>,
+}
+
+impl AccountSecrets {
+    /// Convert the internal tuple into an `AccountFile`
+    ///
+    /// If no name is present, a new one is generated based on the current time
+    /// and the index in
+    pub fn as_account_files(&self) -> impl Iterator<Item = AccountFileWithName> {
+        self.secrets.iter().enumerate().map(
+            |(idx, (maybe_name, account, secret_key, account_seed))| {
+                let account_file = AccountFile::new(
+                    account.clone(),
+                    Some(*account_seed),
+                    vec![AuthSecretKey::RpoFalcon512(secret_key.clone())],
+                );
+                // avoid empty strings and construct a new one based on the account type
+                let name =
+                    maybe_name.clone().filter(|name| !name.is_empty()).unwrap_or_else(|| {
+                        let account_prefix = account.account_type().to_string().to_lowercase();
+
+                        let now = chrono::Local::now();
+                        let now = now.format("%Y%m%d_%H%M%S");
+
+                        format!("{account_prefix}_account{idx:02}__{now}.mac")
+                    });
+                AccountFileWithName { name, account_file }
+            },
+        )
+    }
+}
+
+/// Specify a set of faucets and wallets with assets for easier test depoyments
+///
+/// Notice: Any faucet must be declared _before_ it's use in a wallet/regular account.
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct GenesisConfig {
+    version: u32,
+    timestamp: u32,
+    wallet: Vec<WalletConfig>,
+    faucet: Vec<FaucetConfig>,
+}
+
+impl GenesisConfig {
+    /// Read the genesis accounts from a toml formatted string
+    ///
+    /// Notice: It will generate the specified case during [`fn into_state`].
+    pub fn read_toml(toml_str: &str) -> Result<Self, Error> {
+        let me = toml::from_str::<Self>(toml_str)?;
+        Ok(me)
+    }
+
+    /// Convert the in memory representation into the new genesis state
+    ///
+    /// Notice: Generates keys and returns the secret keys, hence this is not sane to be used
+    /// for production environments. There, you want to generate the keys externally.
```

Comment:
I'm not sure this statement is true. Why shouldn't they use the keys generated here?